src/api/cubrid_log.c

```diff
@@ -0,0 +1,1317 @@
+    {
+      return CUBRID_LOG_INVALID_USER_ARR_SIZE;
+    }
+
+  g_extraction_user = (char **) malloc (sizeof (char *) * arr_size);
+  if (g_extraction_user == NULL)
+    {
+      return CUBRID_LOG_FAILED_MALLOC;
+    }
+
+  for (i = 0; i < arr_size; i++)
+    {
+      g_extraction_user[i] = strdup (user_arr[i]);
+    }
+
+  g_extraction_user_count = arr_size;
+
+  return CUBRID_LOG_SUCCESS;
+}
+
+static int
+cubrid_log_connect_server_internal (char *host, int port, char *dbname)
+{
+  unsigned short rid = 0;
+
+  char *recv_data = NULL;
+  int recv_data_size;
+
+  int reason;
+
+  CSS_QUEUE_ENTRY *queue_entry;
+  int err_code;
+
+  g_conn_entry = css_make_conn (INVALID_SOCKET);
+  if (g_conn_entry == NULL)
+    {
+      CUBRID_LOG_ERROR_HANDLING (CUBRID_LOG_FAILED_CONNECT);
+    }
+
+  if (css_common_connect
+      (host, g_conn_entry, DATA_REQUEST, dbname, strlen (dbname) + 1, port, g_connection_timeout, &rid, true) == NULL)
+    {
+      CUBRID_LOG_ERROR_HANDLING (CUBRID_LOG_FAILED_CONNECT);
+    }
+
+  css_queue_user_data_buffer (g_conn_entry, rid, sizeof (int), (char *) &reason);
+
+  if (css_receive_data (g_conn_entry, rid, &recv_data, &recv_data_size, g_connection_timeout * 1000) != NO_ERRORS)
+    {
+      CUBRID_LOG_ERROR_HANDLING (CUBRID_LOG_FAILED_CONNECT);
+    }
+
+  if (recv_data == NULL || recv_data_size != sizeof (int))
+    {
+      CUBRID_LOG_ERROR_HANDLING (CUBRID_LOG_FAILED_CONNECT);
+    }
+
+  reason = ntohl (*(int *) recv_data);
+
+  if (reason != SERVER_CONNECTED)
+    {
+      CUBRID_LOG_ERROR_HANDLING (CUBRID_LOG_FAILED_CONNECT);
+    }
+
+  if (recv_data != NULL && recv_data != (char *) &reason)
+    {
+      free_and_init (recv_data);
+    }
+
+  return CUBRID_LOG_SUCCESS;
+
+cubrid_log_error:
+
+  if (recv_data != NULL && recv_data != (char *) &reason)
+    {
+      free_and_init (recv_data);
+    }
+
+  queue_entry = css_find_queue_entry (g_conn_entry->buffer_queue, rid);
+  if (queue_entry != NULL)
+    {
+      queue_entry->buffer = NULL;
+      css_queue_remove_header_entry_ptr (&g_conn_entry->buffer_queue, queue_entry);
+    }
+
+  if (g_conn_entry != NULL)
+    {
+      css_free_conn (g_conn_entry);
+      g_conn_entry = NULL;
+    }
+
+  return err_code;
+}
+
+static int
+cubrid_log_send_configurations (void)
+{
+  unsigned short rid = 0;
+
+  OR_ALIGNED_BUF ((OR_INT_SIZE * 5)) a_request;
+  OR_ALIGNED_BUF (OR_INT_SIZE) a_reply;
+  char *request = OR_ALIGNED_BUF_START (a_request), *ptr;
+  char *reply = OR_ALIGNED_BUF_START (a_reply);
+  int request_size = OR_ALIGNED_BUF_SIZE (a_request);
+  int reply_size = OR_ALIGNED_BUF_SIZE (a_reply), reply_code;
+
+  char *recv_data = NULL;
+  int recv_data_size;
+
+  CSS_QUEUE_ENTRY *queue_entry;
+  int err_code;
+
+  ptr = or_pack_int (request, g_max_log_item);
+  ptr = or_pack_int (ptr, g_extraction_timeout);
+  ptr = or_pack_int (ptr, g_all_in_cond);
+  ptr = or_pack_int (ptr, g_extraction_user_count);
+  ptr = or_pack_int (ptr, g_extraction_table_count);
+
```

Comment:
I think 'g_extraction_user' and 'g_extraction_table' are required to be packed also